### DISCUSSION TITLE
keep header visible with left drawer open on small screens

### DISCUSSION THREAD

**Discussion by andrisi** [quasarframework/quasar#15560] [General - Components / Directives / etc]
I'd like to keep the header visible on small screens even if the left drawer is open, as the header has the button to open and close the drawer and it feels like the natural UX-wise to keep it there. I change the icon to open the drawer to "opened" state (another icon) then the drawer is open, so I need the user to be able to tap on it to close.

Clicking outside the drawer works, but its not intuitive, or not the only intuitive action I would like to handle. Ideally the drawer content would start below the header automatically, the top not being "hidden" under the header. This would also make it work as it is on desktop (feature parity), so this would be nics at least as an option.

I found two threads discussing it, but no solutions (the image is from one of them).

https://github.com/quasarframework/quasar/issues/6813
https://forum.quasar-framework.org/topic/3071/how-can-i-do-drawer-on-the-small-screen-under-header

![image](https://user-images.githubusercontent.com/1255809/225036267-a1163c64-06e2-4218-8f31-f4af132dc5a1.png)

**Comment by jrphifer03** [comment]
Not sure if this answers your question.  This may work.  
`<q-layout` view="hHh Lpr `lFf">`

**Reply by andrisi** [reply]
Me neither, did you test it on a small screen? I use the "hHh LpR fFf", so essentially the same, but it does not.

**Reply by jrphifer03** [reply]
sorry.  I misread your question.

**Comment by pdanpdan** [comment]
https://codepen.io/pdanpdan/pen/KKxRdej?editors=1010
`:breakpoint="0"`

**Comment by andrisi** [comment]
Thanks @pdanpdan ! Almost perfext. There is only a small issue: when the left drawer is open, in overlay mode. When I start to scroll the content under it, and the address bar slides out upwards in Chrome, then the bottom of the drawer reveals the content under it, and then quickly corrects, but only after I release my tapping. If possible would be nice to speed up this catch-up cycle. In any case, thanks for the quick help!

![Screenshot_20230315-112920](https://user-images.githubusercontent.com/1255809/225296567-2f492ba7-09e0-40b7-9712-af61e7bb0325.jpg)

**Reply by pdanpdan** [reply]
https://quasar.dev/options/screen-plugin#introduction

`setDebounce` to a smaller value

**Reply by andrisi** [reply]
bravo! (to this, and Quasar in general)

**Comment by andrisi** [comment]
Just a very small sideeffect: if I tap outside the drawer to close it, the event is not "eaten" by this action, but handled by the content below, so in this case, the click to close the drawer also opens a list item. This does not happen without the `show-if-above :breakpoint="0" overlay` options. I know there is `@click.stop` but I don't know what to attach it to.

**Reply by metalsadman** [reply]
you can attach nothing

**Reply by andrisi** [reply]
now I get why, but I can deal with this problem by ignoring the click in some cases on the list's code

**Comment by andrisi** [comment]
Seems like screen.setDebounce() only has effect on the scrolling issue if I set `show-if-above` but I don't need the drawer to be open on startup.
